modules/hal_logger/src/hal_log_flash.hpp

```diff
@@ -0,0 +1,71 @@
+/* *******************************************************************************************
+ * Copyright (c) 2023 by RobotPatient Simulators
+ *
+ * Authors: Victor Hogeweij
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction,
+ *
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software
+ * is furnished to do so,
+ *
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ *
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+***********************************************************************************************/
+#ifndef HAL_LOGTRANSPORT_FLASH_HPP
+#define HAL_LOGTRANSPORT_FLASH_HPP
+#include <hal_log_base.hpp>
+
+namespace hal::log {
+// The amount of characters reserved for the filepath
+inline constexpr uint8_t kMaxFilePathSize = 100;
+
+class FlashLogger : public Logger {
+ public:
+  explicit FlashLogger(LoggerSettings* communicationSettings)
+      : Logger(communicationSettings) {
+    if (communicationSettings->CommMethod == communicationMethod::Flash) {
+      fatfs_ = communicationSettings->CommHandle.FlashHandle.FatHandle;
+      const char* FilePath =
+          communicationSettings->CommHandle.FlashHandle.FilePath;
+      // We unfornately need to memset our filepath buffer.
+      // Because if we do not memset, our filepath will get garbled characters at the end :/
+      memset(FilePath_, '\0', kMaxFilePathSize);
+      memcpy(FilePath_, FilePath, strlen(FilePath));
+    }
+  }
+  void init();
```

Comment:
I understand why you didn't put any documentation here, but if you change the implementation of a virtual function you should still document it.